consensus/hotstuff/cruisectl/block_time_controller.go

```diff
@@ -34,16 +34,18 @@ type TimedBlock struct {
 // epochInfo stores data about the current and next epoch. It is updated when we enter
 // the first view of a new epoch, or the EpochSetup phase of the current epoch.
 type epochInfo struct {
-	curEpochFirstView     uint64
-	curEpochFinalView     uint64    // F[v] - the final view of the epoch
-	curEpochTargetEndTime time.Time // T[v] - the target end time of the current epoch
-	nextEpochFinalView    *uint64
+	curEpochFirstView      uint64
+	curEpochFinalView      uint64  // F[v] - the final view of the current epoch
+	curEpochTargetEndTime  uint64  // T[v] - the target end time of the current epoch, represented as Unix Time [seconds]
+	nextEpochFinalView     *uint64 // the final view of the next epoch
+	nextEpochTargetEndTime *uint64 // the target end time of the next epoch, represented as Unix Time [seconds]
 }
 
 // targetViewTime returns τ[v], the ideal, steady-state view time for the current epoch.
 // For numerical stability, we avoid repetitive conversions between seconds and time.Duration.
 // Instead, internally within the controller, we work with float64 in units of seconds.
 func (epoch *epochInfo) targetViewTime() float64 {
+	// TODO we need to pass TargetDuration via service event as well :(
```

Comment:
I just noticed that `TargetEndTime` is not sufficient to power the PID controller :disappointed:. We also need the duration of the epoch in seconds in order to compute the steady-state optimal view time (`tau`). Previously, we had hard-coded 1 week and used that. 

I don't think there is a way to keep the same process variable without also passing the duration in seconds via service event. Not a huge deal, but unfortunately will require another round of smart contract changes. cc @AlexHentschel